src/issueFinder.ts

```diff
@@ -22,20 +22,24 @@ export class IssueFinder {
         labels: string[],
         repositoryIdentifiers: string[]
     ): Promise<Issue[]> {
-        let result;
-        for (let label of labels) {
-            result = await this.fetcher.fetchMatchingIssues(token, label, repositoryIdentifiers);
-        }
-
-        if (result === undefined) {
-            return [];
-        }
-
-        const issues = result.edges.map((edge) => {
-            const node = edge.node;
-            const issue = this.convertFromGitHubIssue(node);
-            return issue;
+        const arrayOfIssues = labels.map(async (label) => {
+            const result = await this.fetcher.fetchMatchingIssues(
+                token,
+                label,
+                repositoryIdentifiers
+            );
+            const issuesArray = result.edges.map((edge) => {
+                const node = edge.node;
+                const issue = this.convertFromGitHubIssue(node);
+
+                return issue;
+            });
+
+            return issuesArray;
         });
+
+        const issues = (await Promise.all(arrayOfIssues)).flat();
```

Comment:
I might extract a const here to get the result of Promise.all, and then have const issues= xxx.flat() as a separate line.